src/module/auto-savings/installation.ts

```diff
@@ -0,0 +1,49 @@
+import { Address, encodeAbiParameters } from 'viem'
+import { Module } from '../types'
+import { AUTO_SAVINGS_ADDRESS } from './constants'
+
+type Params = {
+  tokens: Address[]
+  configs: {
+    percentage: number
+    vault: Address
+    sqrtPriceLimitX96: number
+  }[]
+}
+
+export const getInstallAutoSavingsExecutor = ({
+  tokens,
+  configs,
+}: Params): Module => {
+  return {
+    module: AUTO_SAVINGS_ADDRESS,
+    data: encodeAbiParameters(
+      [
+        { internalType: 'address[]', name: '_tokens', type: 'address[]' },
+        {
+          components: [
+            { internalType: 'uint16', name: 'percentage', type: 'uint16' },
+            { internalType: 'address', name: 'vault', type: 'address' },
+            {
+              internalType: 'uint128',
+              name: 'sqrtPriceLimitX96',
+              type: 'uint128',
+            },
+          ],
+          internalType: 'struct AutoSavings.Config[]',
+          name: '_configs',
+          type: 'tuple[]',
+        },
+      ],
+      [
+        tokens,
+        configs.map((config) => ({
+          ...config,
+          sqrtPriceLimitX96: BigInt(config.sqrtPriceLimitX96),
```

Comment:
could it make sense to just use bigint type for this?

Reply:
Yep, Fixed